modules/renter/repair.go

```diff
@@ -336,6 +330,12 @@ func (r *Renter) managedScheduleChunkRepair(rs *repairState, chunkID chunkID, ch
 		missingPieces = missingPieces[:len(usefulWorkers)]
 	}
 
+	// Erasure code the pieces.
+	pieces, err := file.erasureCode.Encode(chunkData, missingPieces)
+	if err != nil {
+		return build.ExtendErr("unable to erasure code chunk data", err)
+	}
```

Comment:
That isn't guaranteed to happen anyway unless you force a GC cycle. I don't think the complexity is worth it.

What we really want is a shared buffer that `Encode` will write to, e.g.:

```go
var erasureBuffer [][]byte // initialized to necessary capacity
var erasureLock sync.Mutex

func repair() {
    erasureLock.Lock()
    pieces := erasureCode.Encode(chunkData, erasureBuffer)
    upload(pieces)
    erasureLock.Unlock()
}
```
This would put a hard limit on the amount of memory allocated for erasure coding, and it would also eliminate the need to `nil` unused pieces.